routes/stats.py

```diff
@@ -175,48 +171,196 @@ def stats() -> Union[Response, str]:
+            author_result = top_authors(session=session)
+
+            # Scraping and parsing stats
+            chart_data = chart_stats(session=session, num_days=days)
+
+            return render_template(
+                "stats.html",
+                title="Tölfræði",
+                articles_result=articles_result,
+                articles_total=articles_total,
+                gender_result=gender_result,
+                gender_total=gender_total,
+                author_result=author_result,
+                scraped_chart_data=json.dumps(chart_data["scraped"]),
+                scraped_avg=int(round(chart_data["scraped"]["avg"])),
+                parsed_chart_data=json.dumps(chart_data["parsed"]),
+                parsed_avg=round(chart_data["parsed"]["avg"], 1),
+            )
+    except Exception as e:
+        logging.error(f"Error rendering stats page: {e}")
+        return Response(f"Error: {e}", status=500)
+
+
+_DEFAULT_QUERY_STATS_PERIOD = 30
+_MAX_QUERY_STATS_PERIOD = 30
+
+
+def query_stats_data(
+    session=None, num_days: int = _DEFAULT_QUERY_STATS_PERIOD
+) -> Dict[str, Any]:
+    """Return all data for query stats dashboard."""
+    today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+
+    labels = []
+    query_count_data = []
+
+    # Get query count for each day
+    # We change locale to get localized date weekday/month names
+    with changedlocale(category="LC_TIME"):
+        for n in range(0, num_days):
+            days_back = num_days - n - 1
+            start = today - timedelta(days=days_back)
+            end = today - timedelta(days=days_back - 1)
+
+            # Generate date label
+            dfmtstr = "%a %-d. %b"
+            labels.append(start.strftime(dfmtstr))
+
+            # Get query count for day
+            q = list(QueryCountQuery.period(start, end, enclosing_session=session))
+            query_count_data.append(q[0][0])
+
+    query_avg = sum(query_count_data) / num_days
+
+    start = today - timedelta(days=num_days)
+    end = datetime.utcnow()
+
+    # Query types
+    res = QueryTypesQuery.period(
+        start=start,
+        end=end,
+        enclosing_session=session,
+    )
+
+    # This function is used to ensure that all the query
+    # types have a fixed, unique color on the pie chart.
+    def _hexcolor4string(s: str) -> str:
+        """Generate a hex color from a string."""
+        hash = md5(s.encode("utf-8")).hexdigest()
+        hash_values = (
+            hash[:8],
+            hash[8:16],
+            hash[16:24],
+        )
+        rgb = tuple(int(value, 16) % 256 for value in hash_values)
+        return "#%02x%02x%02x" % rgb
+
+    query_types_data = {
+        "labels": [k[1] for k in res],
+        "datasets": [
+            {
+                "data": [k[0] for k in res],
+                "backgroundColor": [_hexcolor4string(k[1]) for k in res],
+            }
+        ],
+    }
+
+    # Client types
+    _CLIENT_COLORS = {
```

Comment:
Again, a good way to allocate these is to do a HSL value with constant S and L but with the Hs equally spaced around a 360 deg circle (and converted to 0..255).

Reply:
Nah, these I want hardcoded to reflect the standard color scheme assigned to the different platforms, i.e. light blue for iPhone and the green color for Android.